src/column.rs

```diff
@@ -0,0 +1,105 @@
+use core::marker::PhantomData;
+use core::ptr::NonNull;
+
+use alloc::vec::Vec;
+
+use crate::entities::EntityMeta;
+use crate::{Archetype, Component, ComponentError, Entity, MissingComponent};
+
+/// Column of a single component
+pub struct Column<'a, T> {
+    entities: &'a [EntityMeta],
+    archetypes: &'a [Archetype],
+    archetype_column_offsets: Vec<Option<usize>>,
+    _marker: PhantomData<T>,
+}
+
+impl<'a, T: Component> Column<'a, T> {
+    pub(crate) fn new(
+        entities: &'a [EntityMeta],
+        archetypes: &'a [Archetype],
+        archetype_column_offsets: Vec<Option<usize>>,
+        _marker: PhantomData<T>,
+    ) -> Self {
+        Self {
+            entities,
+            archetypes,
+            archetype_column_offsets,
+            _marker,
+        }
+    }
+
+    /// Get this column's component from entity if it exist
+    pub fn get(&self, entity: Entity) -> Result<&T, ComponentError> {
+        unsafe {
+            let meta = self
+                .entities
+                .get(entity.id as usize)
+                .expect("no such entity");
```

Comment:
We should return `ComponentError::NoSuchEntity` rather than panicking here. Additionally, we need to check the `generation` of `entity`. Finally, please narrow the scope of `unsafe` as much as possible; e.g. `[T]::get` is perfectly safe.